qa-tests-backend/src/test/groovy/DeclarativeConfigTest.groovy

```diff
@@ -467,6 +467,178 @@ oidc:
         }
     }
 
+    @Tag("BAT")
+    def "Check orphaned declarative resources are correctly handled"() {
+        when:
+
+        createDefaultSetOfResources(CONFIGMAP_NAME, DEFAULT_NAMESPACE)
+
+        then:
+        // Retry this multiple times, with a pause of 60 seconds.
+        // It may take some time until a) the config map contents are mapped within the pod b) the reconciliation
+        // has been triggered.
+        // If the tests are flaky, we have to increase this value.
+        withRetry(5, 60) {
+            def response = IntegrationHealthService.getDeclarativeConfigHealthInfo()
+            // Expect 6 integration health status for the created resources and one for the config map.
+            assert response.integrationHealthCount == CREATED_RESOURCES + 1
+            for (integrationHealth in response.integrationHealthList) {
+                assert integrationHealth.hasLastTimestamp()
+                assert integrationHealth.getErrorMessage() == ""
+                assert integrationHealth.getStatus() == Status.HEALTHY
+            }
+        }
+
+        when:
+        deleteConfigMapValue(CONFIGMAP_NAME, DEFAULT_NAMESPACE, PERMISSION_SET_KEY)
+
+        then:
+        // Verify the integration health for the permission set is unhealthy and contains an error message.
+        // The errors will be surface after at least three consecutive occurrences, hence we need to retry multiple
+        // times here.
+        withRetry(5, 60) {
+            def response = IntegrationHealthService.getDeclarativeConfigHealthInfo()
+            def permissionSetHealth = response.getIntegrationHealthList().find {
+                it.getName().contains(PERMISSION_SET_KEY)
+            }
+            assert permissionSetHealth
+            assert permissionSetHealth.getErrorMessage()
```

Comment:
Shall we check for a specific keyword within the error message here, i.e. `referenced by another object` so it's clear what error we expect?

Applies to all error messages essentially.